src/tcp/state.ml

```diff
@@ -151,7 +151,13 @@ module Make(Time:Mirage_time.S) = struct
       | Fin_wait_2 _, Recv_fin ->
         Lwt.async (fun () -> timewait t time_wait_time);
         Time_wait
-      | Closing a, Recv_ack b -> if diffone b a then Time_wait else Closing a
+      | Closing a, Recv_ack b -> 
+        if diffone b a then 
+          begin 
+            Lwt.async (fun () -> timewait t time_wait_time);
```

Comment:
I agree the fix is correct.
It seems to always be an error to just transition into Time_wait without arming the timer, maybe we should change `timewait` to actually perform all of it and already return `Time_wait`